nodejs/eks/cmd/provider/cni.ts

```diff
@@ -137,7 +149,10 @@ function computeVpcCniYaml(cniYamlText: string, args: VpcCniInputs): string {
         env.push({ name: "AWS_VPC_ENI_MTU", value: "9001" });
     }
     if (args.image) {
-        daemonSet.spec.template.spec.containers[0].image = args.image.toString();
```

Comment:
note: previously the logic depended on the ordering of containers which is error-prone. I've updated the code to do a scan instead.